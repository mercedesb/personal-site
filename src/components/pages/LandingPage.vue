<template>
  <main :key="`${landingPage.id}_mainContent`" v-if="landingPage.title" class='LandingPage'>
    <PageHeader
      :color="landingPage.color"
      :icon="iconSvg"
    >
      <template slot='decorativeHeader'>
        <h1>{{ landingPage.title }}</h1>
      </template>
      <template slot='titleHeader'>
        <h2>{{ landingPage.preamble }}</h2>
      </template>
    </PageHeader>
    <div class="PageContent" v-if="landingPage.mainContent">
      <ParseMarkdown :source="landingPage.mainContent" :collapsible="false" :collapsibleTag="'h3'" :collapsedByDefault="true" />
    </div>
    <!-- <ContactForm v-if="landingPage.showContact" /> -->
    <BlogListContainer v-if="landingPage.showBlogPosts" :color="landingPage.color" :page="parseInt(pageNumber)" />
    <TalkList v-if="landingPage.showTalks" />
    <div :key="`${landingPage.id}_ctaLinks`" v-if="ctaLinks.length" class='FlexContainer PageContent PageContent--wide LandingPage-ctaContainer'>
      <CTALink
        v-for="ctaLink in ctaLinks"
        :key="ctaLink.id"
        v-bind="ctaLink"
      />
    </div>
  </main>
</template>

<script>
import PageHeader from '../PageHeader.vue'
import CTALink from '../CTALink.vue'
import BlogListContainer from '../BlogListContainer.vue'
import TalkList from '../TalkList.vue'
import ParseMarkdown from '../ParseMarkdown.vue'

export default {
  components: {
    PageHeader,
    CTALink,
    BlogListContainer,
    TalkList,
    ParseMarkdown
  },
  props: {
    landingPage: Object,
    iconSvg: String,
    ctaLinks: Array,
    pageNumber: Number
  }
}
</script>

<style lang="scss">
@import '../../assets/styles/variables.scss';

.LandingPage {
  @include inner-page-content;

  &-ctaContainer {
    margin-top: $large-spacing;
  }
}

</style>
