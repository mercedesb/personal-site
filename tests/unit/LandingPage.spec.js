import LandingPage from "@/components/pages/LandingPage.vue";
import { TestUtility } from "./TestUtility";

describe("LandingPage", () => {
  let component;

  const initialProps = {
    landingPage: {
      ...TestUtility.landingPages["about"]
    },
    ctaLinks: []
  };

  const shallow = propsData =>
    TestUtility.shallow(LandingPage, {
      propsData: {
        ...initialProps,
        ...propsData
      }
    });

  describe("Snapshots", () => {
    describe("without an iconSvg", () => {
      it("matches snapshot", () => {
        component = shallow({ iconSvg: "" });
        expect(component.element).toMatchSnapshot();
      });
    });

    describe("with an iconSvg", () => {
      it("matches snapshot", () => {
        component = shallow({ iconSvg: "<svg></svg>" });
        expect(component.element).toMatchSnapshot();
      });
    });

    describe("without ctaLinks", () => {
      it("matches snapshot", () => {
        component = shallow();
        expect(component.element).toMatchSnapshot();
      });
    });

    describe("with ctaLinks", () => {
      it("matches snapshot", () => {
        const ctaLinks = [
          {
            title: "title",
            iconSvg: "<svg></svg>",
            url: "/url"
          },
          {
            title: "title",
            iconSvg: "<svg></svg>",
            url: "/url",
            color: "yellow"
          }
        ];
        component = shallow({ ctaLinks });
        expect(component.element).toMatchSnapshot();
      });
    });
  });
});
