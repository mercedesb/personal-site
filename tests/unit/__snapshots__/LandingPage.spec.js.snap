// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Snapshots with an iconSvg matches snapshot 1`] = `
<main
  class="LandingPage"
>
  <pageheader-stub
    color="blue"
    icon=<svg></svg>
  >
    <template>
      <h1>
        About
      </h1>
    </template>
     
    <template>
      <h2>
        
      </h2>
    </template>
  </pageheader-stub>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
</main>
`;

exports[`LandingPage Snapshots with ctaLinks matches snapshot 1`] = `
<main
  class="LandingPage"
>
  <pageheader-stub
    color="blue"
  >
    <template>
      <h1>
        About
      </h1>
    </template>
     
    <template>
      <h2>
        
      </h2>
    </template>
  </pageheader-stub>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <div
    class="FlexContainer PageContent PageContent--wide LandingPage-ctaContainer"
  >
    <ctalink-stub
      color="brown"
      iconsvg=<svg></svg>
      title="title"
      url="/url"
    />
    <ctalink-stub
      color="yellow"
      iconsvg=<svg></svg>
      title="title"
      url="/url"
    />
  </div>
</main>
`;

exports[`LandingPage Snapshots without an iconSvg matches snapshot 1`] = `
<main
  class="LandingPage"
>
  <pageheader-stub
    color="blue"
    icon=""
  >
    <template>
      <h1>
        About
      </h1>
    </template>
     
    <template>
      <h2>
        
      </h2>
    </template>
  </pageheader-stub>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
</main>
`;

exports[`LandingPage Snapshots without ctaLinks matches snapshot 1`] = `
<main
  class="LandingPage"
>
  <pageheader-stub
    color="blue"
  >
    <template>
      <h1>
        About
      </h1>
    </template>
     
    <template>
      <h2>
        
      </h2>
    </template>
  </pageheader-stub>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
</main>
`;
